rskj-core/src/test/java/co/rsk/peg/BridgeSupportFlyoverTest.java

```diff
@@ -0,0 +1,2189 @@
+        );
+
+        Address retiringFederationAddress = PegTestUtils.getFastBridgeAddressFromRedeemScript(
+            bridgeConstants,
+            retiringFederation.getRedeemScript(),
+            Sha256Hash.wrap(fastBridgeDerivationHash.getBytes())
+        );
+
+        BtcTransaction btcTx = new BtcTransaction(bridgeConstants.getBtcParams());
+        btcTx.addOutput(valueToSend, retiringFederationAddress);
+
+        btcTx.addInput(
+            Sha256Hash.wrap(fastBridgeDerivationHash.getBytes()),
+            0, ScriptBuilder.createInputScript(null, new BtcECKey())
+        );
+
+        List<Sha256Hash> hashes = new ArrayList<>();
+        hashes.add(btcTx.getHash());
+        // Create header and PMT
+        byte[] bits = new byte[1];
+        bits[0] = 0x3f;
+        PartialMerkleTree pmt = new PartialMerkleTree(bridgeConstants.getBtcParams(), bits, hashes, 1);
+        Sha256Hash merkleRoot = pmt.getTxnHashAndMerkleRoot(new ArrayList<>());
+        co.rsk.bitcoinj.core.BtcBlock registerHeader = new co.rsk.bitcoinj.core.BtcBlock(
+            bridgeConstants.getBtcParams(),
+            1,
+            PegTestUtils.createHash(2),
+            merkleRoot,
+            1,
+            1,
+            1,
+            new ArrayList<>()
+        );
+        int height = 1;
+        // simulate blockchain
+        mockChainOfStoredBlocks(
+            btcBlockStore,
+            registerHeader,
+            height + bridgeConstants.getBtc2RskMinimumAcceptableConfirmations(),
+            height
+        );
+
+        InternalTransaction rskTx = new InternalTransaction(
+            Keccak256.ZERO_HASH.getBytes(),
+            0,
+            0,
+            null,
+            null,
+            null,
+            lbcAddress.getBytes(),
+            null,
+            null,
+            null,
+            null
+        );
+
+        co.rsk.core.Coin preCallLbcAddressBalance = repository.getBalance(lbcAddress);
+
+        BigInteger result = bridgeSupport.registerFastBridgeBtcTransaction(
+            rskTx,
+            btcTx.bitcoinSerialize(),
+            height,
+            pmt.bitcoinSerialize(),
+            derivationArgumentsHash,
+            userRefundBtcAddress,
+            lbcAddress,
+            lpBtcAddress,
+            true
+        );
+
+        // if transaction went OK following assertions should be met
+        if (result.signum() == 1) {
+            co.rsk.core.Coin expectedBalance = preCallLbcAddressBalance.add(co.rsk.core.Coin.fromBitcoin(
+                valueToSend
+            ));
+            co.rsk.core.Coin postCallLbcAddressBalance = repository.getBalance(lbcAddress);
+
+            // assert the new balance of the Lbc address is correct
+            Assert.assertEquals(
+                expectedBalance,
+                postCallLbcAddressBalance
+            );
+
+            // when RSKIP293 is active, the method setFastBridgeRetiringFederationInformation should be called twice.
```

Comment:
```suggestion
            // when RSKIP293 is active, the method markFastBridgeFederationDerivationHashAsUsed should be called twice.
```
That's what you meant, right?

Reply:
Yes.

Reply:
Done